packages/next/src/server-build.ts

```diff
@@ -1131,6 +1131,23 @@ export async function serverBuild({
     });
   }
 
+  if (appDir) {
```

Comment:
`appDir` is `null` unless `appPathRoutesManifest` is truthy, so you could check `appPathRoutesManifest` and then you don't need the `|| {}` below.